docs/src/assets/0_pregenerate.jl

```diff
@@ -16,3 +16,42 @@ let
              xlabel="Ecut", ylabel="energy absolute error", label="")
     savefig(p, "convergence_study_ecut.png")
 end
+
+let
+    include("../../../../examples/pseudopotentials.jl")
+
+    function run_scf(Ecut, psp, tol)
+        println("Ecut = $Ecut")
+        println("----------------------------------------------------")
+        a = 5.0
+        lattice   = a * Matrix(I, 3, 3)
+        atoms     = [ElementPsp(:Li, psp=psp)]
+        positions = [zeros(3)]
+    
+        model = model_LDA(lattice, atoms, positions; temperature=1e-2)
+        basis = PlaneWaveBasis(model; Ecut=Ecut, kgrid=[8, 8, 8])
+        self_consistent_field(basis; tol)
+    end
+    
+    function converge_Ecut(Ecuts, psp, tol)
+        energies = [run_scf(Ecut, psp, tol/100).energies.total for Ecut in Ecuts]
+        errors = abs.(energies[begin:end-1] .- energies[end])
+        iconv = findfirst(errors .< tol)
+        (Ecuts=Ecuts[begin:end-1], errors, Ecut_conv=Ecuts[iconv])
+    end
+
+    Ecuts = 20:4:96
+    tol   = 1e-3
+
+    conv_hgh = converge_Ecut(Ecuts, psp_hgh, tol)
+    conv_upf = converge_Ecut(Ecuts, psp_upf, tol)
+
+    println("HGH: $(onv_hgh.Ecut_conv)")
+    println("UPF: $(conv_upf.Ecut_conv)")
+
+    plt = plot(xlabel="Ecut [Ha]", ylabel="Error [Ha]")
+    plot!(plt, conv_hgh.Ecuts, conv_hgh.errors, label="HGH")
+    plot!(plt, conv_upf.Ecuts, conv_upf.errors, label="PseudoDojo UPF")
```

Comment:
Show convergence on a logplot, thicker lines.

Reply:
Done!

Reply:
![image](https://user-images.githubusercontent.com/75646713/197182649-23006491-12fc-4098-8b03-328b62feeb0c.png)

Reply:
Very fishy, why is there such a big plateau? Is it because of the tolerance? If so can you just not display that part of the curve?

Reply:
Otherwise very interesting! I knew the hgh were kind of hard but the difference is huge!

Reply:
Yes it's the SCF tolerance. By default in this example `tol/100` where `tol` is the convergence we are aiming for (so 1e-3).

Reply:
Or the other option is to select the SCF tolerance smaller (like 1e-9) ... since we are pregenerating this, that should not be an issue.

Reply:
Hmm I'm not 100% sure it was the SCF tolerance. I ran with 1e-10 on the SCF, and the plateau was still there. It would make sense for it to be the SCF tolerance, so I'm not sure what's happening

Reply:
I changed up the convergence study a bit to calculate a reference energy at `140 Ha` cutoff then go through a range of Ecuts until one is converged w.r.t. that.

![image](https://user-images.githubusercontent.com/75646713/197196604-70de0a25-cb5d-4429-9ad1-303cb18b8a5b.png)

Reply:
I also changed the tolerance to `1 meV / atom`

Reply:
36 and 128(!) Ha, for reference

Reply:
ok. The plateau might also be related to the fineness of the numerical grid of the projectors?

Reply:
Exactly, that was my concern and the thing I wanted to make sure of. @azadoks can you run with an even lower tol?

Reply:
![image](https://user-images.githubusercontent.com/75646713/197223090-25fd42a4-d952-4fbe-9de1-58c3f03cbe57.png)

Reply:
:scream: That does look like a numerical grid issue. Do you by any chance have a graph of |V(q)| and |beta(q)| in semilog scale at the ready (so we can see at which rate it decays)? Otherwise I can take a look

Reply:
I can probably get it in a few mins.

Reply:
Yup, it's a grid issue. I'm opening a separate issue, it shouldn't hold back the merge

Reply:
![image](https://user-images.githubusercontent.com/75646713/197228793-3fe3fb5e-761e-4523-8e34-b4055fb84852.png)
![image](https://user-images.githubusercontent.com/75646713/197229523-4289f3ee-262d-4e1b-bcd1-3b942b59b2fd.png)

Reply:
![image](https://user-images.githubusercontent.com/75646713/197229850-ea8c9ddf-07b5-4da4-b432-ff850861569b.png)

Reply:
Note the potential seems stagnant around q=10, which is around Ecut=50, so that's consistent.